src/theory/evaluator.cpp

```diff
@@ -475,20 +487,47 @@ EvalResult Evaluator::evalInternal(
         }
         case kind::DIVISION:
         case kind::DIVISION_TOTAL:
+        case kind::INTS_DIVISION:
+        case kind::INTS_DIVISION_TOTAL:
+        case kind::INTS_MODULUS:
+        case kind::INTS_MODULUS_TOTAL:
         {
           Rational res = results[currNode[0]].d_rat;
           bool divbyzero = false;
+          Kind k = currNodeVal.getKind();
+          bool isReal = (k == kind::DIVISION || k == kind::DIVISION_TOTAL);
+          bool isMod =
+              (k == kind::INTS_MODULUS || k == kind::INTS_MODULUS_TOTAL);
           for (size_t i = 1, end = currNode.getNumChildren(); i < end; i++)
           {
             if (results[currNode[i]].d_rat.isZero())
             {
-              Trace("evaluator")
-                  << "Division by zero not supported" << std::endl;
-              divbyzero = true;
-              results[currNode] = EvalResult();
-              break;
+              if (k == kind::DIVISION_TOTAL || k == kind::INTS_DIVISION_TOTAL
+                  || k == kind::INTS_MODULUS_TOTAL)
+              {
+                res = Rational(0);
```

Comment:
Do I understand correctly that with this you are interpreting div by 0 as 0?

Reply:
Yes, the `_TOTAL` versions have that semantics.